### ISSUE TITLE
CEC Detection for the TV

### ISSUE THREAD

**Issue by superzanti** [hyperion-project/hyperion.ng#708]
#### What problem does this feature solve?

The lights typically will stay on when the TV turns off unless all other devices are turned off too.
Often times this isn't the case since one may leave something on to update, or to just play music in the background.

#### What does the proposed API look like?

An extra 'CEC Detection' button for the V4L2 USB Capture.
This would assume that the pi has an HDMI connection to the TV and can read the TV's CEC.

![CEC and Width/Height Additions](https://i.imgur.com/19z9HXX.png)

#### How should this be implemented in your opinion?

I think a `_cecDetectionEnabled` variable could be added to the V4L2Grabber.cpp.
A check on this just before line 1083 where it does the check for `_signalDetectionEnabled`.

Some rough code as follows:
```
    if (_cecDetectionEnabled)
    {
        CEC::ICECAdapter *g_parser;
        g_parser = CEC::LibCecInitialise(&g_config);
        if(g_parser)
        {
            auto pwr = g_parser->GetDevicePowerStatus(addr);
            if (pwer != CEC::CEC_POWER_STATUS_ON)
            {
		        emit newFrame(image);
            }
            else
            {
		    	Info(_log, "Power to TV is OFF");
            }
        }
    }
```

#### Are you willing to work on this yourself?

Yes, but with guidance. There are many things that I don't understand. However, a developer familiar with the code should be able to implement it pretty fast with the use of libcec as noted above.

**Comment by superzanti** [comment]
For anyone looking to do this in the meantime, [here is a handy script](https://hyperion-project.org/threads/bash-script-to-turn-off-on-leds-with-hdmi-cec.4384/)

**Comment by T-IOOO** [comment]
Wow! This would really be awsome!

**Comment by T-IOOO** [comment]
I am willing to test this out, I have made the changes in the cpp-file but since I am novice I dont know what I need to do further for the changes to take effect. Could you please make me a ”dummy” explanation on what I need to do next?

**Comment by superzanti** [comment]
I didn't get too far into this.

But you may look up some examples on the libcec and see if you can get something working that just outputs a 1 or a 0 to the terminal when the screen is on or off. Do this and implementation is pretty much done.

**Comment by m-seker** [comment]
@superzanti @T-IOOO  Are you still eager to give it a try ? If yes, you can checkout #877 branch and compile. Feedback appreciated as I don't have chance to try with different hardware.

**Comment by T-IOOO** [comment]
I Gave up a while ago because the cec-utils wouldn't recognize my TV. The cause of it seemed to be the HDMI splitter. I will try it out as soon as I can.

**Comment by Lord-Grey** [comment]
Closed by #877
